src/test/java/igym/controllers/GymControllerTest.java

```diff
@@ -88,63 +91,53 @@ void testCreateGymWithDuplicateName() throws Exception {
     @Test
     @DisplayName("should return 400 Bad Request when name is null")
     void testCreateGymWithNullName() throws Exception {
-
-        Gym gym = new Gym(null);
-
         mockMvc.perform(post("/api/gyms")
                 .contentType(MediaType.APPLICATION_JSON)
-                .content(objectMapper.writeValueAsString(gym)))
+                .content(objectMapper.writeValueAsString(new Gym())))
                 .andExpect(status().isBadRequest())
                 .andExpect(jsonPath("$.message").value("Validation error"))
-                .andExpect(jsonPath("$.errors.name").value("Name cannot be blank"));
+                .andExpect(jsonPath("$.errors").value("Name cannot be blank"));
 
         verify(gymService, never()).createGym(any(Gym.class));
     }
 
     @Test
     @DisplayName("should return 400 Bad Request when name an empty string")
     void testCreateGymWithEmptyStringName() throws Exception {
-
-        Gym gym = new Gym("");
-
         mockMvc.perform(post("/api/gyms")
                 .contentType(MediaType.APPLICATION_JSON)
-                .content(objectMapper.writeValueAsString(gym)))
+                .content(objectMapper.writeValueAsString(new Gym(""))))
                 .andExpect(status().isBadRequest())
-                .andExpect(jsonPath("$.message").value("Validation error"))
-                .andExpect(jsonPath("$.errors.name").value("Name must be between 3 and 50 characters"));
+                .andExpect(jsonPath("$.errors", hasSize(2))) // Ensure two errors exist
+                .andExpect(jsonPath("$.errors", containsInAnyOrder(
+                        "Name cannot be blank",
+                        "Name must be between 3 and 50 characters")));
```

Comment:
good solution I have applied this in my test and solved the nondeterministic problem